openshift/openshift-copies.go

```diff
@@ -394,17 +387,17 @@ func validateAuthInfo(authInfoName string, authInfo clientcmdAuthInfo) []error {
 
 		if len(authInfo.ClientCertificate) != 0 {
 			clientCertFile, err := os.Open(authInfo.ClientCertificate)
-			defer clientCertFile.Close()
 			if err != nil {
 				validationErrors = append(validationErrors, errors.Errorf("unable to read client-cert %v for %v due to %v", authInfo.ClientCertificate, authInfoName, err))
 			}
+			defer clientCertFile.Close()
```

Comment:
Why is this moved? If golangci-lint doesn't complain, then it doesn't belong in this PR.